### PR TITLE
Updates to `Cleanup_Msdb` procedure

### PR DESCRIPTION
**Pull request by amtwo**
* Make dynamic SQL safe from SQL Injection (via object name)
* Make dynamic SQL safe for unicode object names
* Switch to use `Cleanup_BackupHistory` procedure, rather than msdb default. This version is performance optimized.

### CODE REVIEW

File: stored-procedures/dbo.Cleanup_Msdb.sql [hunk 1]
```diff
@@ -32,7 +32,7 @@ DECLARE @RetainDate datetime2;
 SELECT @RetainDate = DATEADD(DD,-1*@RetainDays,SYSDATETIME());
```
Comment (AaronBertrand) [inline stored-procedures/dbo.Cleanup_Msdb.sql]: I would do DATEADD(DAY, -@RetainDays

File: stored-procedures/dbo.Cleanup_Msdb.sql [hunk 2]
```diff
@@ -45,8 +45,10 @@ EXEC msdb.dbo.sysmail_delete_mailitems_sp @sent_before = @RetainDate;
     
 
 --And do index maintenance
-DECLARE @Sql nvarchar(max) = ''
-SELECT @Sql = @Sql + 'ALTER INDEX [' + i.name + '] ON [msdb].[' + s.name + '].[' + t.name + '] REORGANIZE;' + CHAR(10)
+-- if you aren't doing index maintenance on msdb, we'll force it here
+-- sometimes this prevents significant space bloat, esp on older versions
+DECLARE @Sql nvarchar(max) = N''
+SELECT @Sql = @Sql + N'ALTER INDEX ' + QUOTENAME(i.name) + N' ON [msdb].' + QUOTENAME(s.name) + N'.' + QUOTENAME(t.name) + N' REORGANIZE;' + CHAR(10)
```
Comment (AaronBertrand) [inline stored-procedures/dbo.Cleanup_Msdb.sql:51]: And if you're in here I'd change sp_executesql to sys.sp_executesql

**Review by AaronBertrand — CHANGES_REQUESTED** [review]
Just two minor improvements

File: stored-procedures/dbo.Cleanup_Msdb.sql (hunk 1 above)
Comment (amtwo) [reply stored-procedures/dbo.Cleanup_Msdb.sql]: you would.

File: stored-procedures/dbo.Cleanup_Msdb.sql (hunk 2 above)
Comment (amtwo) [reply stored-procedures/dbo.Cleanup_Msdb.sql:51]: Good catch! 👍

File: stored-procedures/dbo.Cleanup_Msdb.sql (hunk 1 above)
Comment (amtwo) [reply stored-procedures/dbo.Cleanup_Msdb.sql]: Switched to use `DAY`

**Pull request merged**